From: Takashi Iwai <tiwai@suse.de>
Subject: kABI workaround for hci_chan amp field addition
Patch-mainline: Never, kABI workaround
References: CVE-2021-33034 bsc#1186111

The newly added amp flag in hci_chan struct breaks kABI.
As the new field fits into a hole, we just need to wrap it with
__GENKSYMS__ ifndef as usual.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/net/bluetooth/hci_core.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/net/bluetooth/hci_core.h
+++ b/include/net/bluetooth/hci_core.h
@@ -540,7 +540,9 @@ struct hci_chan {
 	struct sk_buff_head data_q;
 	unsigned int	sent;
 	__u8		state;
+#ifndef __GENKSYMS__
 	bool		amp;
+#endif
 };
 
 struct hci_conn_params {
